internal/shared/version/version.go

```diff
@@ -17,8 +17,26 @@ var (
 	}
 )
 
+// isUnset returns true when the provided string should be treated as an
+// "unset" value. Builds that inject ldflags such as "-X var=" will set the
+// variable to the empty string, which previously prevented the runtime build
+// information gathered via debug.ReadBuildInfo from populating the field. For
+// the purposes of version reporting we treat both the empty string and the
+// literal "unknown" as unset.
+func isUnset(s string) bool {
+	return s == "" || s == "unknown"
+}
+
 func String() string {
-	return fmt.Sprintf("version: %q, commit: %q, date: %q, state: %q", version, gitCommit, commitDate, repoState)
+	return fmt.Sprintf("version: %q, commit: %q, date: %q, state: %q",
+		valueOrUnknown(version), valueOrUnknown(gitCommit), valueOrUnknown(commitDate), valueOrUnknown(repoState))
```

Comment:
Could we format it in a way that is easier to read?
Aiming for maintainability 

```
func String() string {
	return fmt.Sprintf("version: %q, commit: %q, date: %q, state: %q",
		valueOrUnknown(version),
		valueOrUnknown(gitCommit),
		valueOrUnknown(commitDate),
		valueOrUnknown(repoState))
}
```

Reply:
yep, doing in next commit